### PR TITLE
JavaScript code refactoring

### PR DESCRIPTION
**Pull request by elemoine**
This PR makes some refactoring to the JavaScript code.

### CODE REVIEW

File: thinkhazard/static/js/report.js [hunk 1]
```diff
@@ -16,21 +19,27 @@
...
+  var timeoutId = -1;
   $(window).resize(function() {
     if (timeoutId != -1) {
       clearTimeout(timeoutId);
     }
-    timeoutId = setTimeout(addMap, 500);
+    timeoutId = setTimeout(function() {
+      addMap(hazardType);
+      timeoutId = -1;
+    }, 500);
```
Comment (pgiraud) [inline thinkhazard/static/js/report.js]: Do we really need to affect a "-1" value to the timeoutId?
What if we call `clearTimeout` each time?

Comment (elemoine) [inline thinkhazard/static/js/report.js]: I don't know what `clearTimeout` does when you pass it an invalid timeoutId value.

Comment (pgiraud) [inline thinkhazard/static/js/report.js]: It fails silently:
```javascript
var foo;
clearTimeout(foo);
```

Comment (elemoine) [inline thinkhazard/static/js/report.js]: Ok, so unconditionally calling `clearTimeout` should work.

Comment (elemoine) [inline thinkhazard/static/js/report.js]: Changed as requested.

**Pull request merged**